platform/openide.nodes/src/org/openide/nodes/PropertySupport.java

```diff
@@ -87,6 +92,162 @@ static <T> T cast(Class<T> c, Object o) {
         }
     }
 
+    /**
+     * Creates a "virtual" property where getter and setter are backed by the
+     * provided {@link Supplier} and {@link Consumer} functional interfaces.
+     * @param <T> the type of the property
+     * @param name the name of the property
+     * @param valueType the type of the property
+     * @param displayName the display name of the property, can be {@code null}.
+     * @param shortDescription the short description (used in tooltip) of the property, can be {@code null}.
+     * @param supplier the getter functional interface, can be {@code null} for write-only properties.
+     * @param consumer the setter functional interface, can be {@code null} for read-only properties.
+     *
+     * @since 7.62
+     * @return a {@link PropertySupport} instance where getter and setter are
+     *         backed by the provided functional interfaces.
+     */
+    public static <T> PropertySupport<T> readWrite(String name, Class<T> valueType, String displayName, String shortDescription, Supplier<T> supplier, Consumer<T> consumer) {
+        return new FunctionalProperty<>(name, valueType, displayName, shortDescription, supplier, consumer);
+    }
+
+    /**
+     * Creates a "virtual" property where getter and setter are backed by the
+     * provided {@link Supplier} and {@link Consumer} functional interfaces.
+     * @param <T> the type of the property
+     * @param name the name of the property
+     * @param valueType the type of the property
+     * @param displayName the display name of the property, can be {@code null}.
+     * @param supplier the getter functional interface, can be {@code null} for write-only properties.
+     * @param consumer the setter functional interface, can be {@code null} for read-only properties.
+     *
+     * @since 7.62
+     * @return a {@link PropertySupport} instance where getter and setter are
+     *         backed by the provided functional interfaces.
+     */
+    public static <T> PropertySupport<T> readWrite(String name, Class<T> valueType, String displayName, Supplier<T> supplier, Consumer<T> consumer) {
```

Comment:
Not a review objection, rather a thought to consider.

There are 3 overloads for each flavour (readOnly, readWrite) that serve rather as a convenience methods that do not require the caller to use `null` as argument (displayName, sdescription). Is that necessary ? I mean given that the result is `PropertySupport` = `Node.Property` = `FeatureDescriptor`, then anyone can call `setShortDescription` or `setDisplayName` afterwards (since it's in FeatureDescriptor). Or just pass `null` to the most-args factory method.
The `null`-passing technique is used to make Functional propsupport readonly or  writeonly anyway.

Another alternative, to make the construction fluent - and with an effect to other potential PropertySupport subclasses would be to add 
`public final PropertySupport withShortDescription(String s) { setShortDescription(s); return this;}` and the same for display name ... number of characters typed is +- same with the current state, but can be configured in one expression.

Reply:
Thanks for the feedback, I think I'd go with the fluent construction. That's a good idea! Thanks!